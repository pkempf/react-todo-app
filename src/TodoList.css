.TodoList {
  top: 10px;
  margin: auto;
  padding: 10px;
}
