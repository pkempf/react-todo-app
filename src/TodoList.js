import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import "./TodoList.css";

const TodoList = () => {
  const [todos, setTodos] = useState([]);
  const addTodo = (todoObj) => {
    setTodos((todos) => [...todos, todoObj]);
  };
  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  const todosToRender = todos.map((todo) => (
    <Todo
      key={todo.id}
      id={todo.id}
      task={todo.task}
      handleRemove={removeTodo}
    />
  ));

  return (
    <div className="TodoList">
      <NewTodoForm createTodo={addTodo} />
      <hr />
      {todosToRender}
    </div>
  );
};

export default TodoList;
