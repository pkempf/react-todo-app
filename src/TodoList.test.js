import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TodoList from "./TodoList";

const addTodo = (todoList, task = "abcdefg") => {
  const taskInput = todoList.getByLabelText("Task:");
  const submitButton = todoList.getByText("Create Task");
  fireEvent.change(taskInput, { target: { value: task } });
  fireEvent.click(submitButton);
};

it("renders without crashing", () => {
  render(<TodoList />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("adds a todo", () => {
  const todoList = render(<TodoList />);

  // the todo shouldn't be there yet
  expect(todoList.queryByText("abcdefg")).not.toBeInTheDocument();

  // add the todo
  addTodo(todoList);

  // make sure the task and remove button were added
  expect(todoList.getByText("abcdefg")).toBeInTheDocument();
  expect(todoList.getByText("X")).toBeInTheDocument();

  // make sure our form was reset
  expect(todoList.getByLabelText("Task:")).toHaveValue("");
});

it("deletes a todo", () => {
  const todoList = render(<TodoList />);
  addTodo(todoList);

  // make sure the todo is there to start with
  expect(todoList.getByText("abcdefg")).toBeInTheDocument();

  // click the X to remove it
  fireEvent.click(todoList.getByText("X"));

  // todo should be gone now
  expect(todoList.queryByText("abcdefg")).not.toBeInTheDocument();
});
