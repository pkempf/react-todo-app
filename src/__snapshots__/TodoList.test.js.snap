// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="NewTodoForm"
    >
      <form>
        <div>
          <label
            for="task"
          >
            Task:
          </label>
          <input
            id="task"
            name="task"
            type="text"
            value=""
          />
        </div>
        <button
          id="newTaskButton"
        >
          Create Task
        </button>
      </form>
    </div>
    <hr />
  </div>
</DocumentFragment>
`;
